working/extension_structs/overview.md

```diff
@@ -0,0 +1,543 @@
+contain a single entry in their primary constructor.
+
+
+```dart
+extension struct Data(int x);
+extension struct GenericData<T>(T y);
+```
+
+
+#### Alternatives
+
+There are various alternatives for the core "struct" feature described in the
+previous section, and any alternative choice made there would impact this
+feature correspondingly.  Ignoring that, several alternative syntaxes for
+extension structs are on the table.
+
+##### Alternative modifiers
+
+The choice of the keyword "extension" is intended to leverage users existing
+intuitions about how extension methods work: that is, that they are largely
+statically dispatched.  It's not clear to me that this intuition actually
+carries over, however. There are various alternatives to "extension".
+
+We could use "static", reflecting the static nature of the dispatch. 
+
+```dart
+static struct Data(int x);
+static struct GenericData<T>(T y);
+```
+
+We could use "view" reflecting the fact that we are presenting a "view" on an object.
+
+```dart
+view struct Data(int x);
+view struct GenericData<T>(T y);
+```
+
+We could use "typedef" reflecting the fact that we are largely defining a static
+construct:
+
+```dart
+typedef struct Data(int x);
+typedef struct GenericData<T>(T y);
+```
+
+We could use "type", or "new type".
+
+
+### Fields and primary constructors
+
+Primary constructor lists for extension structs are exactly identical to those
+of normal structs, with the restriction that they may contain only one entry.
+
+Examples:
+
+```dart
+extension struct Data(int x);
+extension struct GenericData<T>(T y);
+```
+
+### Members of extension structs
+
+An extension struct may contain static and instance member definitions in the
+same way as a class, with exactly the same syntactic resolution.
+
+For scope resolution purposes, entries in the primary constructor list are
+treated exactly as if they were defined as instance members on the extension
+struct.
+
+It is an error for an extension struct to define a field as a member.
+
+It is an error if an extension struct declares an abstract member, unless a
+member of the same name and kind is available on the unique field of the
+extension struct, and the type of the abstract member is a supertype of the type
+of the corresponding member in the unique field.
+
+### Abstract extension structs
+
+Extension structs may not be be marked abstract.
+
+### Extending extension structs
+
+An extension struct may not be extended.
+
+### Implementing extension structs
+
+It is an error to implement an extension struct.  Extension structs do not
+define interfaces.
+
+### Using extension structs as a bound
+
+It is an error to use the type defined by an extension struct as a bound on a
+generic type parameter.
+
+### Extension structs implementing interfaces
+
+Extension structs may implement interfaces if and only if each implemented
+interface type is a subtype of the type of the unique field in the primary
```

Comment:
DBC: should it say "supertype of the type of the unique field" instead?

Reply:
yep, thanks!